matter_persistence/sql/models.py

```diff
@@ -0,0 +1,80 @@
+import datetime
+from enum import Enum
+
+from pydantic import BaseModel
+from sqlalchemy import Column, DateTime, Float
+from sqlalchemy.ext.declarative import declared_attr
+
+
+class SortMethodModel(Enum):
+    ASC = "asc"
+    DESC = "desc"
+
+
+class BaseDBModel:
+    def init_fields(self):
+        self.created_at = datetime.datetime.now(tz=datetime.UTC)
+        self.created_at_timestamp = self.created_at.timestamp()
+
+    def assign(self, base_model: BaseModel, update: bool = False):
+        for key, value in base_model.model_dump().items():
+            ignored_fields = [
+                "created_at",
+                "created_at_timestamp",
+                "deleted_at",
+                "updated_at",
+            ]
+            if key in ignored_fields:
+                continue
+
+            if hasattr(self, key):
+                setattr(self, key, value)
+
+            if update:
+                self.updated_at = datetime.datetime.now(tz=datetime.UTC)
+
+    def as_dict(self):
+        return self.__dict__
+
+    def soft_delete(self):
+        self.deleted_at = datetime.datetime.now(tz=datetime.UTC)
+
+    @declared_attr
+    def created_at(cls):
+        return Column(DateTime(timezone=True), default=None, nullable=False)
+
+    @declared_attr
+    def created_at_timestamp(cls):
+        return Column(Float, default=None, nullable=False)
```

Comment:
I thought you wanted to remove this

Reply:
Yes, I was just a bit slow with my commits.